StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerSK2Tests.swift

```diff
@@ -131,4 +133,122 @@ class TrialOrIntroPriceEligibilityCheckerSK2Tests: StoreKitConfigTestCase {
             expect(receivedEligibility.status) == expected[product]
         }
     }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForProductIsEligible() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "com.revenuecat.monthly_4.99.1_week_intro"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let status = try await withCheckedThrowingContinuation({ (continuation: ContinuationStatusResult) in
```

Comment:
I think we don't even need an explicit type here, i.e.: 
```suggestion
        let status = try await withCheckedThrowingContinuation({ continuation in
```

Reply:
same thinking in other places where this happens, I don't think having the explicit type adds much in this case

Reply:
it was failing to infer the type, but I think adding a type to the variable declaration itself instead of in the closure makes more sense 👍 (then I can also remove the typealias, that was to keep the line length down for swiftlint)